### PR TITLE
 📖add minikube info into dev doc

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

update doc to include minikube
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/dev-quickstart.md [hunk 1]
```diff
@@ -67,7 +71,7 @@ cert-manager-webhook-6b57b9b886-5cbnp      1/1     Running   0          32s
 As a cluster API~(CAPI) provider, CAPN requires core components of CAPI to be setup. We need to deploy the unreleased version of CAPI for `v1alpha4` API support.
 
 ```console
-git clone git@github.com:kubernetes-sigs/cluster-api.git
+git clone https://github.com/kubernetes-sigs/cluster-api.git
```
Comment (neolit123) [inline docs/dev-quickstart.md:74]: no need to switch to https here.

File: docs/dev-quickstart.md [hunk 2]
```diff
@@ -269,4 +279,4 @@ kubectl --kubeconfig kubeconfig get all -A
 
 ```shell
 kind delete cluster --name=capn
-```
\ No newline at end of file
+```
```
Comment (neolit123) [inline docs/dev-quickstart.md:282]: this removes the empty line at EOF; better to leave it.

**Review by neolit123 — COMMENTED** [review]
added a couple of comments.

including minikube in the quickstart seems fine; but ideally this document does not want to maintain examples for all the local cluster solutions.

File: docs/dev-quickstart.md (hunk 1 above)
Comment (jichenjc) [reply docs/dev-quickstart.md:74]: actually, not everyone use git (ssh) ... but I can leave it as it is

**Review by gyliu513 — COMMENTED** [review]
```console
kind load docker-image docker.io/virtualcluster/apiserver-v1.16.2:latest --name=capn
kind load docker-image docker.io/virtualcluster/controller-manager-v1.16.2:latest --name=capn
kind load docker-image docker.io/virtualcluster/etcd-v3.4.0:latest --name=capn
```

This also need minikube mapping.

**Comment by gyliu513** [comment]
Sorry, last comment, https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/docs/dev-quickstart.md#clean-up also needs updated.

**Comment by jichenjc** [comment]
> 
> 
> Sorry, last comment, https://github.com/kubernetes-sigs/cluster-api-provider-nested/blob/main/docs/dev-quickstart.md#clean-up also needs updated.

sure, updated.

**Review by christopherhein — CHANGES_REQUESTED** [review]
It's hard to apply these comments as suggestions but can we make it so that only the command line commands are contained within the backticks for all of these and break minikube vs kind commands into separate ones?

**Comment by jichenjc** [comment]
> 
> 
> It's hard to apply these comments as suggestions but can we make it so that only the command line commands are contained within the backticks for all of these and break minikube vs kind commands into separate ones?

updated, thanks~

**Review by christopherhein — COMMENTED** [review]
/lgtm
/approve

🎉

**Pull request merged**